CMakeLists.txt

```diff
@@ -1,117 +1,166 @@
 cmake_minimum_required(VERSION 3.10.2)
-project(point_cloud_transport)
 
-find_package(catkin REQUIRED COMPONENTS
-  cras_cpp_common
-  cras_topic_tools
-  dynamic_reconfigure
-  message_generation
-  nodelet
-  pluginlib
-  roscpp
-  sensor_msgs
-  topic_tools
-)
+set(CMAKE_CXX_STANDARD 17)
 
-find_package(Boost REQUIRED)
+project(point_cloud_transport)
+
+find_package(ament_cmake_ros REQUIRED)
 
-catkin_python_setup()
+find_package(message_filters REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(sensor_msgs REQUIRED)
 
-generate_dynamic_reconfigure_options(cfg/NoConfig.cfg)
+include_directories(
+  include
+)
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME} raw_${PROJECT_NAME}
-  CATKIN_DEPENDS cras_cpp_common cras_topic_tools dynamic_reconfigure message_filters message_runtime nodelet roscpp sensor_msgs topic_tools
+include(FetchContent)
+fetchcontent_declare(
```

Comment:
I intentionally did not use this approach in the ROS 1 version. It breaks offline builds (unless fetchcontent has become somehow much more clever).

Reply:
Offline build is eg. when you're in the field with a robot without internet connection and need to rebuild the workspace. Cmake fetchcontent prevents this.

Submodules are friendlier in this regard, however I never liked them and I'm not sure how tools like vcstool work with them.

Plainly copying such a "simple" library seemed to me like the best approach.